my-app/src/app/modules/courses-page/components/courses/courses.component.ts

```diff
@@ -1,18 +1,73 @@
-import {Component, Input, Output, EventEmitter, ChangeDetectionStrategy} from '@angular/core';
+import {Component, OnInit, OnDestroy} from '@angular/core';
 
 import {ICourse} from "@shared/";
 
+import {isEqual} from "lodash";
+
+import {Subscription} from "rxjs";
+
+import {FilterPipe} from "../../pipes";
+import {CoursesHandlerService} from "../../services/courses-handler.service";
+
 @Component({
   selector: 'app-courses',
   templateUrl: './courses.component.html',
   styleUrls: ['./courses.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush
+  providers: [FilterPipe]
 })
-export class CoursesComponent {
-  @Input() course: ICourse | undefined;
-  @Output() deleteCourse = new EventEmitter<number>();
+export class CoursesComponent implements OnInit, OnDestroy {
+  public visibleCourses: ICourse[] = [];
+  public searchValue: string = '';
+
+  private _subscriptions$: Subscription = new Subscription();
+  private existingCourses: ICourse[] = [];
+
+  constructor(private filterPipe: FilterPipe,
+              private coursesHandlerService: CoursesHandlerService) {
+  }
+
+  ngOnInit(): void {
+    this.getListOfCourses();
+  }
+
+  ngOnDestroy(): void {
+    this._subscriptions$?.unsubscribe();
+  }
+
+  private getListOfCourses(): void {
+    this._subscriptions$.add(this.coursesHandlerService.getList()
+      .subscribe(existingCourses => {
+        this.visibleCourses = structuredClone(existingCourses);
+        this.existingCourses = structuredClone(existingCourses);
+      }));
+  }
+
+  public onDeleteCourse(courseId: number): void {
+    const foundCourse: ICourse | undefined = this.coursesHandlerService.getItemById(courseId);
+
+    if (!foundCourse) {
+      console.error("Course wasn't found");
+      return;
+    }
+
+    if (foundCourse
```

Comment:
After the previous `if`, `foundCourse` will definetely be truthy.

Reply:
Yes, thank you for catching it